runtime/tests/sharedclassapi/sharedClasses.c

```diff
@@ -0,0 +1,641 @@
+ static char* msg24 = "Correct number of caches found. Expected at least %d caches, found %d. ";
+ static char* msg25 = "Incorrect number of caches found. Expected at least %d, found %d";
+ static char* msg26 = "Starting iterating over caches from VM Init event\n";
+ static char* msg27 = "Iterating over caches: ERROR";
+ static char* msg28 = "Iterating over caches: SUCCESS";
+ static char* msg29 = "No value found for option '%s'";
+ static char* msg30 = "Parsing option string '%s'";
+ static char* msg31 = "The default cache directory will be used";
+ static char* msg32 = "cache Directory to be searched is '%s'";
+ static char* msg33 = "Invalid value specified for expectedCacheCount. Expected an integer, got '%s'";
+ static char* msg34 = "Number of caches found will not be checked";
+ static char* msg35 = "Number of caches found will be checked. Expected value is %d";
+ static char* msg36 = "Command line values for the shared classes cache directory will be used";
+ static char* msg37 = "Command line values for the shared classes cache directory will be ignored";
+ static char* msg38 = "Will attempt to delete the caches after querying them";
+ static char* msg39 = "Caches will not be deleted after querying them";
+ static char* msg40 = "Cache prefix was specified as: '%s'. Only caches with this prefix will be deleted";
+ static char* msg41 = "No Cache prefix was specified. All caches will be deleted";
+ static char* msg42 = "onload processing complete\n\n";
+ static char* msg43 = "Unexpected return code '%d' from '%s'\n\n";
+
+ #ifdef ZOS
+   #pragma convlit(resume)
+ #endif
+
+ /**
+  * Returns a newly allocated copy of original. If length is
+  * -1, then, copies the whole of original. If length is > 0
+  * and less then length of original, then only that number
+  * of chars will be copied. The length of the returned string
+  * will be 'length' + the null terminator
+  *
+  * Returns null if malloc fails.
+  *
+  */
+ char* copyString(char* original, int length) {
+
+     char* newCopy = NULL;
+
+     if (length == -1) {
+         newCopy = malloc(strlen(original) + 1);
+         if (newCopy == NULL) {
+             return NULL;
+         }
+         strcpy(newCopy, original);
+         return newCopy;
+     }
+
+     newCopy = malloc(length + 1);
+     if (newCopy == NULL) {
+         return newCopy;
+     }
+
+     strncpy(newCopy, original, length);
+
+     newCopy[length] = '\0';
+     return newCopy;
+ }
+
+
+ /**
+  * This should be called as a pair with freeConvertedString
+  *
+  * Takes an ascii string.
+  * On Z/OS, copies the string, converts to ebcdic, and returns
+  * the new, copied string.
+  *
+  * On other platforms, returns the passed in string.
+  *
+  */
+ char* copyAndConvert(char* asciiString) {
+
+     #ifdef ZOS
+
+         char* copiedString;
+         copiedString = copyString(asciiString, -1);
+         __atoe(copiedString);
+         return copiedString;
```

Comment:
Please join the declaration and the assignment and add the missing `NULL` check:
```c
	char *copiedString = copyString(asciiString, -1);
	if (NULL != copiedString) {
		__atoe(copiedString);
	}
	return copiedString;
```